databricks/koalas/indexes.py

```diff
@@ -961,6 +961,81 @@ def max(self):
 
         return result if len(result) > 1 else result[0]
 
+    def delete(self, loc):
+        """
+        Make new Index with passed location(-s) deleted.
+
+        .. note:: this API can be pretty expensive since it is based on
+             a global sequence internally.
+
+        Returns
+        -------
+        new_index : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10])
+        >>> kidx
+        Int64Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete(0)
+        Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete([0, 1, 2, 3, 10, 11])
+        Int64Index([4, 2, 4, 4, 2, 2], dtype='int64')
+        """
+        if isinstance(self, MultiIndex):
+            raise NotImplementedError("currently doesn't support for MultiIndex")
+        if not is_list_like(loc):
+            if not isinstance(self, str) and abs(loc) >= len(self):
+                raise IndexError(
+                    "index {} is out of bounds for axis 0 with size {}"
+                    .format(loc, len(self)))
+            loc = [loc]
+        loc = [int(item) for item in loc]
+        loc = [item if item >= 0 else len(self) + item for item in loc]
+
+        # we need below temporal column 'index_value_columns'
+        # since '_InternalFrame.attach_default_index' will be failed
+        # when if self._scol has name of '__index_level_0__'
+        index_value_column = '__index_value__'
+
+        sdf = self._internal.sdf
+        sdf = sdf.select(self._scol.alias(index_value_column))
+
+        sdf = _InternalFrame.attach_default_index(
+            sdf, default_index_type='distributed-sequence')
+        # sdf here looks like below
+        # +-----------------+---------------+
+        # |__index_level_0__|__index_value__|
+        # +-----------------+---------------+
+        # |                1|             10|
+        # |                6|              4|
+        # |                3|              8|
+        # |                5|              2|
+        # |                9|              2|
+        # |                4|              4|
+        # |                8|              2|
+        # |                7|              4|
+        # |               10|             10|
+        # |               11|             10|
+        # |                2|              9|
+        # |                0|             10|
+        # +-----------------+---------------+
+
+        conds = [F.col(SPARK_INDEX_NAME_FORMAT(0)) != idx_loc for idx_loc in loc]
```

Comment:
How about using `isin`?

Reply:
Thanks for the comments and sorry for delay